etl/steps/data/garden/happiness/2024-06-09/happiness.meta.yml

```diff
@@ -0,0 +1,81 @@
+# NOTE: To learn more about the fields, hover over their names.
+definitions:
+  common:
+    processing_level: major
+    presentation:
+      topic_tags:
+        - Happiness & Life Satisfaction
+
+
+# Learn more about the available fields:
+# http://docs.owid.io/projects/etl/architecture/metadata/reference/
+dataset:
+  update_period_days: 365
+
+
+tables:
```

Comment:
If possible, it would be good to retain the links to the previous reports ([like here](https://github.com/owid/etl/blob/41f9b1d7f40fdf19fd8270cb79d03237d5e6708a/etl/steps/data/garden/happiness/2023-03-20/happiness.meta.yml#L18)), as this dataset is a combination of many annual. I'm not sure what would be the best place to do this, though.

This information is stored in each of the Snapshots of the source files, but these do not use the Origin format. It could be a bit of a headache to fix, I'd be curious to hear what @pabloarosado thinks about it.

Reply:
Are we expecting to keep using previous versions of the report every year? In that case, maybe it's worth moving the old step to the new metadata format. If the snapshot doesn't exist anymore, we could just change the snapshot dvc file, and the garden meta yml file (and possibly some minor things in the code).

Reply:
Thanks for the review @pabloarosado - indeed I think it would be good to move the old data set to origins as well. I will follow the steps you laid out - but what do you mean by "if the snapshot doesn't exist anymore"? the old snapshots from 2012-2023 are still under etl/snapshots/happiness

Reply:
Good idea! What I meant is that the data provider's data may not be available anymore. If that's the case, you will not be able to execute the snapshot .py file. However, even then, you can simply modify the content of the .dvc file, and ETL will read the new metadata as expected. There's no need to re-execute the corresponding .py file. Does that make sense?